components/flow-orchestration-framework/org.wso2.carbon.identity.flow.mgt/src/main/java/org/wso2/carbon/identity/flow/mgt/FlowAIService.java

```diff
@@ -0,0 +1,357 @@
+
+/**
+ * Service class for Flow AI operations.
+ * This service provides methods to generate flows using AI, check generation status,
+ * and retrieve generated flow results.
+ */
+public class FlowAIService {
+
+    private static final Log LOG = LogFactory.getLog(FlowAIService.class);
+
+    private static final String FLOW_AI_ENDPOINT = IdentityUtil.getProperty(Constants.FlowAIConstants.FLOW_AI_ENDPOINT);
+    private static final String FLOW_AI_GENERATE_PATH = IdentityUtil.getProperty(
+            Constants.FlowAIConstants.FLOW_AI_GENERATE_PATH);
+    private static final String FLOW_AI_STATUS_PATH = IdentityUtil.getProperty(
+            Constants.FlowAIConstants.FLOW_AI_STATUS_PATH);
+    private static final String FLOW_AI_RESULT_PATH = IdentityUtil.getProperty(
+            Constants.FlowAIConstants.FLOW_AI_RESULT_PATH);
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final FlowAIService INSTANCE = new FlowAIService();
+
+    private FlowAIService() {
+
+    }
+
+    /**
+     * Get FlowAIService instance.
+     *
+     * @return FlowAIService singleton instance.
+     */
+    public static FlowAIService getInstance() {
+
+        return INSTANCE;
+    }
+
+    /**
+     * Generate flow based on the provided request.
+     *
+     * @param flowGenerationRequestDTO Request containing flow type and user query.
+     * @return Response containing operation ID for the generated flow.
+     * @throws FlowMgtClientException If the request is invalid.
+     * @throws FlowMgtServerException If an error occurs while invoking the AI service.
+     */
+    public FlowGenerationResponseDTO generateFlow(FlowGenerationRequestDTO flowGenerationRequestDTO)
+            throws FlowMgtServerException, FlowMgtClientException {
+
+        validateFlowGenerationRequest(flowGenerationRequestDTO);
+        validateAIServiceConfiguration();
+
+        String tenantDomain = IdentityTenantUtil.getTenantDomainFromContext();
+        LOG.debug("Generating flow for tenant: " + tenantDomain + ", flowType: " +
+                flowGenerationRequestDTO.getFlowType());
+
+        try {
+            Map<String, Object> requestBody = buildGenerateFlowRequestBody(flowGenerationRequestDTO);
+            Map<String, Object> response = AIHttpClientUtil.executeRequest(FLOW_AI_ENDPOINT,
+                    FLOW_AI_GENERATE_PATH, HttpPost.class, requestBody);
+
+            if (response == null || response.isEmpty()) {
+                throw FlowMgtUtils.handleServerException(ERROR_CODE_INVOKING_AI_SERVICE, tenantDomain);
+            }
+
+            String operationId = (String) response.get(Constants.FlowAIConstants.OPERATION_ID);
+            if (StringUtils.isBlank(operationId)) {
+                throw FlowMgtUtils.handleServerException(ERROR_CODE_INVOKING_AI_SERVICE, tenantDomain);
+            }
+
+            LOG.debug("Flow generation initiated successfully with operation ID: " + operationId);
```

Comment:
Same as above, let's wrap all the debug logs with debug log check if it has string concatinations